File: app/src/main/java/com/commit451/gitlab/util/ImageUtil.java
package com.commit451.gitlab.util;

import android.net.Uri;

import com.commit451.gitlab.model.api.UserBasic;
import com.commit451.gitlab.model.api.UserFull;

/**
 * Utility for doing various image related things
 */
public class ImageUtil {
    public static Uri getAvatarUrl(UserBasic user, int size) {
        if (user != null) {
            Uri avatarUrl = user.getAvatarUrl();
            if (avatarUrl != null && !avatarUrl.equals(Uri.EMPTY)) {
                return avatarUrl.buildUpon()
                        .appendQueryParameter("s", Integer.toString(size))
                        .build();
            }

            if (user instanceof UserFull) {
                return getAvatarUrl(((UserFull) user).getEmail(), size);
            }
        }

        return getAvatarUrl("", size);
    }

    public static Uri getAvatarUrl(String email, int size) {
        return Gravatar
                .init(email)
                .ssl()
                .size(size)
                .defaultImage(Gravatar.DefaultImage.IDENTICON)
                .build();
    }
}

File: app/src/main/java/com/commit451/gitlab/util/ImageUtil.kt
package com.commit451.gitlab.util

import android.net.Uri
import com.commit451.gitlab.model.api.User

/**
 * Utility for doing various image related things
 */
object ImageUtil {
    fun getAvatarUrl(user: User?, size: Int): Uri {
        if (user != null) {

            if (user.avatarUrl != null) {
                val avatarUrl = Uri.parse(user.avatarUrl)
                if (avatarUrl != null && avatarUrl != Uri.EMPTY) {
                    return avatarUrl.buildUpon()
                            .appendQueryParameter("s", Integer.toString(size))
                            .build()
                }
            }

            val email = user.email
            if (email != null) {
                return getAvatarUrl(email, size)
            }
        }

        return getAvatarUrl(null as? String?, size)
    }

    fun getAvatarUrl(email: String?, size: Int): Uri {
        return Gravatar
                .init(email)
                .ssl()
                .size(size)
                .defaultImage(Gravatar.DefaultImage.IDENTICON)
                .build()
    }
}
